### PR TITLE
Feature | Topic Slider #92

### LINKED ISSUE #92: Feature | Filter Topics on Blogs page
### Where ? 
- https://medium-app-zeta.vercel.app/blogs

### What ?
- Show a list of all topics horizontally like this.

Screenshot 2024-05-24 at 9 59 42 AM

### Backend routes:

- The GET `/api/v1/tag` API endpoint returns a list of all topics which can be used to show all topics
- The GET `api/v1/blog/bulk?tagId=[TAG_ID]` API endpoint returns a list of blogs for the topic with specific TAG_ID

### Objective:
1. Show all topics / tags
2. Filter the blogs if user clicks on any topic / tag.

PS: Reuse existing components wherever possible.

### PR DESCRIPTION
**Pull request by dhairya-create**
closes #92 

<hr /> 

-> Added necessary icons for topic slider
-> Used useSearchParams for getting the tagID of the blogs
->Created custom hook to get the list of the topics

### CODE REVIEW

File: frontend/src/components/BlogsList.tsx
Comment (aadeshkulkarni) [inline frontend/src/components/BlogsList.tsx:1]: @dhairya-create - Any particular reason behind removing the ScrollToTop and the Confetti animations ?

File: frontend/src/hooks/index.ts [hunk 1]
```diff
@@ -81,11 +98,9 @@ export const useBlog = ({ id }: { id: string }) => {
 
   async function fetchBlog() {
     const token = localStorage.getItem('token');
-    /* Temporarily removing this feature
     if (!token) {
       navigate('/signup');
     }
-    */
```
Comment (aadeshkulkarni) [inline frontend/src/hooks/index.ts:88]: This needs to be commented.

File: frontend/src/components/BlogsList.tsx
Comment (dhairya-create) [reply frontend/src/components/BlogsList.tsx:1]: Sorry, I'll fix it it was done by mistake.

Comment (aadeshkulkarni) [reply frontend/src/components/BlogsList.tsx:1]: @dhairya-create - Please fix the issues at the earliest. The feature is important and this ticket has been open for a long time now.

Comment (dhairya-create) [reply frontend/src/components/BlogsList.tsx:1]: Hi @aadeshkulkarni , I have added the topic slider already.

Comment (aadeshkulkarni) [reply frontend/src/components/BlogsList.tsx:1]: Thank you @dhairya-create!  🚀

**Pull request merged**